caching/withCaching.js

```
@@ -35,21 +37,28 @@ const cacheWrapper = (fn, cache, logger, params = {}) => {
     options = {}
   } = params;
   const {
-    expirationSeconds = 300 // 5 mins
+    expirationSeconds = 300, // 5 mins
+    withCompression = false,
+    compressionLib = lzo
   } = options;
 
   return (...args) => {
     const start = Date.now();
     const key = cacheKeySerializer.apply(null, args);
 
     return tryGetFromCache({ cache, key, logger, name })
-      .then(cachedValue => {
+      .then(([cachedValue, didError]) => {
+        if (!cachedValue && didError) {
```

hmm, I was thinking we would still hit the source of truth, but not write to the cache if there was a read error. 

I can see the argument to simply fail fast.  If we want to fail fast should we just rethrow after logging in `tryGetFromCache`?

---

Hm, I misunderstood.

I'm not set on the fail fast, it seems like it could be good _right now_ but in other scenarios it wouldn't be. We eventually want back off behavior that's more nuanced. :thinking:

---

I switched this back to only handling the cache error situation (don't write to cache on error). We can always add a better ATP backoff strategy later.